src/qt/guiutil_font.cpp

```diff
@@ -0,0 +1,596 @@
+                }
+                vecSupported.push_back(*weight);
+            }
+            prevWeight = *weight;
+        }
+        if (vecSupported.empty()) {
+            vecSupported.push_back(QFont::Normal);
+        }
+        return vecSupported;
+    };
+
+    for (const auto& [family, _, __] : AVAILABLE_FONTS) {
+        mapSupportedWeights.insert(std::make_pair(family, supportedWeights(family)));
+    }
+
+    auto getBestMatch = [&](FontFamily fontFamily, QFont::Weight targetWeight) {
+        auto& vecSupported = mapSupportedWeights[fontFamily];
+        auto it = vecSupported.begin();
+        QFont::Weight bestWeight = *it;
+        int nBestDiff = abs(*it - targetWeight);
+        while (++it != vecSupported.end()) {
+            int nDiff = abs(*it - targetWeight);
+            if (nDiff < nBestDiff) {
+                bestWeight = *it;
+                nBestDiff = nDiff;
+            }
+        }
+        return bestWeight;
+    };
+
+    auto addBestDefaults = [&](FontFamily family) -> auto {
+        QFont::Weight normalWeight = getBestMatch(family, g_font_defaults.weight_normal);
+        QFont::Weight boldWeight = getBestMatch(family, g_font_defaults.weight_bold);
+        if (normalWeight == boldWeight) {
+            // If the results are the same use the next possible weight for bold font
+            auto& vecSupported = mapSupportedWeights[family];
+            auto it = std::find(vecSupported.begin(), vecSupported.end(), normalWeight);
+            if (++it != vecSupported.end()) {
+                boldWeight = *it;
+            }
+        }
+        mapDefaultWeights.emplace(family, std::make_pair(normalWeight, boldWeight));
+    };
+
+    for (const auto& [family, _, __] : AVAILABLE_FONTS) {
+        addBestDefaults(family);
+    }
+
+    // Load supported defaults. May become overwritten later.
+    mapWeights = mapDefaultWeights;
+
+    return true;
+}
+
+bool fontsLoaded()
+{
+    return osDefaultFont != nullptr;
+}
+
+void setApplicationFont()
+{
+    if (!fontsLoaded()) {
+        return;
+    }
+
+    std::unique_ptr<QFont> font;
+
+    if (fontFamily == FontFamily::Montserrat) {
+        QString family = fontFamilyToString(FontFamily::Montserrat);
+#ifdef Q_OS_MAC
+        if (getFontWeightNormal() != g_font_defaults.weight_normal) {
+            font = std::make_unique<QFont>(getFontNormal());
+        } else {
+            font = std::make_unique<QFont>(family);
+            font->setWeight(g_font_defaults.weight_normal);
+        }
+#else
+        font = std::make_unique<QFont>(family);
+        font->setWeight(getFontWeightNormal());
+#endif
+    } else {
+        font = std::make_unique<QFont>(*osDefaultFont);
+    }
+
+    font->setPointSizeF(g_font_defaults.size);
+    qApp->setFont(*font);
+
+    qDebug() << __func__ << ": " << qApp->font().toString() << " family: " << qApp->font().family()
+             << ", style: " << qApp->font().styleName() << " match: " << qApp->font().exactMatch();
```

Comment:
nit: qDebug with too many params is a bit messy. Consider:

```suggestion
    qDebug() << strprintf("%s: %s family: %s, style: %s, match: %s", __func__, qApp->font().toString(), qApp->font().family(), qApp->font().styleName(), qApp->font().exactMatch());
```

Reply:
`qstrprintf` implemented in [dash#7068](https://github.com/dashpay/dash/pull/7068) to address this issue, no longer in scope of this PR.